File: 1381-design-a-stack-with-increment-operation/1381-design-a-stack-with-increment-operation.java
class CustomStack {

    private int index = 0;
    private Integer[] stack;
    
    public CustomStack(int maxSize) {
        stack = new Integer[maxSize];
    }
    
    public void push(int x) {
        if (index < stack.length) {
            stack[index] = x;
            index++; 
        } 
    }
    
    public int pop() {
        index--;
        if (index < 0) {
            index = 0;
            return -1;
        }
        int value = stack[index];
        return value;
    }
    
    public void increment(int k, int val) {
        int target = index < k ? index : k;
        for (int i = 0; i < target; i++) {
            stack[i] += val;
        }
    }
}

/**
 * Your CustomStack object will be instantiated and called as such:
 * CustomStack obj = new CustomStack(maxSize);
 * obj.push(x);
 * int param_2 = obj.pop();
 * obj.increment(k,val);
 */

File: 1381-design-a-stack-with-increment-operation/1381-design-a-stack-with-increment-operation.kt
class CustomStack(private val maxSize: Int) {

    private var array: IntArray = IntArray(maxSize)

    fun push(x: Int) {
        val depth = array.filter { it != 0 }.size
        if (depth != maxSize)
            array[depth] = x
    }

    fun pop(): Int {
        val size = array.filter { it != 0 }.size
        return if (size != 0) {
            val value = array[size - 1]
            array[size - 1] = 0
            value
        } else {
            -1
        }
    }

    fun increment(k: Int, `val`: Int) {
        val size = array.filter { it != 0 }.size
        if (size == 0) {
            return
        }
        if (size < k) {
            for (i in 0 until size) {
                array[i] += `val`
            }
        } else {
            for (i in 0 until k) {
                array[i] += `val`
            }
        }
    }

}